jetty-server/src/main/java/org/eclipse/jetty/server/handler/DefaultHandler.java

```diff
@@ -147,13 +148,11 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
                 // Context Path
                 ContextHandler context = (ContextHandler)handlers[i];
 
-                StringBuilder href = new StringBuilder();
-
                 String contextPath = context.getContextPath();
-                href.append(contextPath);
+                String href = UrlEncoded.encodeString(contextPath.substring(1), UTF_8);
```

Comment:
This is the wrong encode to use.  `UrlEncoded` is for parameters not paths

Reply:
Yep, I got that earlier.
The filed issue #3565 and associated PR #3567 are the correct solution, once resolved, that specific solution will be used.